### PR TITLE
Add flux inference pipeline

### PR DESCRIPTION
**Pull request by Victor49152**
# What does this PR do ?

Add flux inference pipeline

**Collection**: 
diffusion

# Changelog 
/nemo/collections/diffusion

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

**Comment by Victor49152** [comment]
This PR is dependent on implementation of media in Mcore. Will need to add the changes to NeMo first or stay as a draft until the Mcore change is merged.

**Comment by ethanhe42** [comment]
@zpx01 can you take a look at the flow matching?

File: nemo/collections/diffusion/vae/autoencoder.py
Comment (ethanhe42) [inline nemo/collections/diffusion/vae/autoencoder.py:1]: is this vae general or specific to flux? maybe a subfolder if it's specific? Also is there things we can reuse from huggingface diffusers?

File: nemo/collections/diffusion/models/dit/dit_layer_spec.py [hunk 1]
```diff
@@ -407,6 +453,225 @@ def forward(
         return output, context
 
 
+class DiTLayer(TransformerLayer):
```
Comment (ethanhe42) [inline nemo/collections/diffusion/models/dit/dit_layer_spec.py:456]: is this the same as the existing DiTLayerWithAdaLN?

**Review by ethanhe42 — CHANGES_REQUESTED** [review]

**Review by ethanhe42 — APPROVED** [review]

**Pull request merged**